tests/components/melcloud/test_config_flow.py

```diff
@@ -0,0 +1,172 @@
+            result["flow_id"],
+            {"username": "test-email@test-domain.com", "password": "test-password"},
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "test-email@test-domain.com"
+    assert result2["data"] == {
+        "username": "test-email@test-domain.com",
+        "token": "test-token",
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+@pytest.mark.parametrize(
+    "error,reason",
+    [(ClientError(), "cannot_connect"), (asyncio.TimeoutError(), "cannot_connect")],
+)
+async def test_form_errors(hass, mock_login, mock_get_devices, error, reason):
+    """Test we handle cannot connect error."""
+    mock_login.side_effect = error
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_USER},
+        data={"username": "test-email@test-domain.com", "password": "test-password"},
+    )
+
+    assert len(mock_login.mock_calls) == 1
+    assert result["type"] == "abort"
+    assert result["reason"] == reason
+
+
+@pytest.mark.parametrize(
+    "error,message",
+    [(401, "invalid_auth"), (403, "invalid_auth"), (500, "cannot_connect")],
+)
+async def test_form_response_errors(
+    hass, mock_login, mock_get_devices, mock_request_info, error, message
+):
+    """Test we handle response errors."""
+    mock_login.side_effect = ClientResponseError(mock_request_info(), (), status=error)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_USER},
+        data={"username": "test-email@test-domain.com", "password": "test-password"},
+    )
+
+    assert result["type"] == "abort"
+    assert result["reason"] == message
+
+
+async def test_import_with_token(hass, mock_login, mock_get_devices):
+    """Test successful import."""
+    with patch(
+        "homeassistant.components.melcloud.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.melcloud.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={"username": "test-email@test-domain.com", "token": "test-token"},
+        )
+
+    assert result["type"] == "create_entry"
+    assert result["title"] == "test-email@test-domain.com"
+    assert result["data"] == {
+        "username": "test-email@test-domain.com",
+        "token": "test-token",
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_token_refresh(hass, mock_login, mock_get_devices):
+    """Re-configuration with existing username should refresh token."""
+    await hass.config_entries.async_add(
```

Comment:
Don't use this. Use `MockConfigEntry.add_to_hass`.
https://github.com/home-assistant/home-assistant/blob/41f3fb291afe6596f276400e8c692e815172f40d/tests/common.py#L692